### ISSUE TITLE
Project Structure Proposal

### ISSUE THREAD

**Issue by tisoft** [ReverseEngineeringDE/SAIC-API-Documentation#64]
Currently we have this project which contains the reverse engineered documentation and the java api and mqtt implementation. Additionally we have https://github.com/ReverseEngineeringDE/iSmart-Gateway-Home-Assistant-Add-on which contains the HA add-on. 

We have a fork from @tosate, who implemented a python integration here: https://github.com/tosate/SAIC-API-Documentation/tree/main/saic-python-mqtt-gateway

I have a fork of the OpenHAB repository, containing the OpenHab plugin here: https://github.com/tisoft/openhab2-addons/tree/saic_api

There might be other language or home automation integrations later.

I would like to propose a new structure:

* We create a new GitHub organization (something like SAIC-Open-API, or something) with the following projects
  * documentation
    * ASN.1 files
    * contents of the docs directory from this project
  * reverse-engineering
    * documentation on how to reverse engineer the iSmart App (the current README file from this project)
    * tools for reverse engineering
      * asn1extractor directory from this project
  * saic-java-api
    * the current java api and mqtt implementation from this project
  * saic-python-api
    * the python implementation from https://github.com/tosate/SAIC-API-Documentation/tree/main/saic-python-mqtt-gateway
  * saic-home-assistant-addon
    * the https://github.com/ReverseEngineeringDE/iSmart-Gateway-Home-Assistant-Add-on project
  * saic-openhab-plugin
    * a fork of the OpenHAB repository, containing the OpenHab plugin, with the goal of upstreaming it back to openhab

What do you think?

Explicitly pinging @ReverseEngineeringDE, @tosate, @mschmiedel, @weidi, @Tonno87 since all of you committed or forked one of the repos.

**Comment by tosate** [comment]
Awesome!

I am currently working on a Python client library for the SAIC iSMART API. This should encapsulate the logic to interact with the API, including ABRP. I will make it available on the [Python Package Index](https://pypi.org/).
My desired repository would be: **saic-python-client**

My MQTT implementation will use the client library. Could be: **saic-python-mqtt-gateway**

If this is working, I can fork https://github.com/home-assistant/core and start working on a Home Assistant integration.

**Comment by tisoft** [comment]
Good idea using -client instead of -api in the language implementations. I have changed the proposal. I am fine with having saic-python-mqtt-gateway in an extra repo. For java, I would like to keep them in one repo for now, since that is easier to handle for me. I think we can decide on a case by case basis what feels more natural for the specific implementation.

**Comment by Tonno87** [comment]
I think it's a good idea to have all these things in one place.

**Comment by weidi** [comment]
Sounds great to me as it would simplify things a lot.

**Comment by tisoft** [comment]
habe die @SAIC-iSmart-API Organisation erstellt und euch eingeladen. Wartet bitte noch, bis ihr da selbst repos erstellt, ich will erst die bestehenden umstrukturieren und da rein packen.
